modules/printable_pdf/src/Plugin/PrintableFormat/PdfFormat.php

```diff
@@ -0,0 +1,205 @@
+    $this->setConfiguration(array(
+      'pdf_generator' => $form_state['values']['pdf_generator'],
+    ));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildContent() {
+   $content = parent::buildContent();
+   $rendered_page = parent::extractLinks(render($content));
+   $this->pdfGenerator->addPage($rendered_page);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function mbuildContent($save_pdf, $filename) {
+    $content = parent::buildContent();
+    $rendered_page = parent::extractLinks(render($content));
+    if($save_pdf) {
+      if(empty($filename)) {
+        $filename=str_replace("/","_",\Drupal::service('path.current')->getPath());
+        $filename=substr($filename, 1);
+      }
+      $this->pdfGenerator->stream(utf8_encode(new Response($rendered_page)), $filename.'.pdf');
+    }
+    else
+      $this->pdfGenerator->send(utf8_encode(new Response($rendered_page)));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getResponse() {
+    $pdf_library = (string)$this->configFactory->get('printable.settings')->get('pdf_tool');
+    $paper_size = (string)$this->configFactory->get('printable.settings')->get('paper_size');
+    $paper_orientation = $this->configFactory->get('printable.settings')->get('page_orientation');
+    $save_pdf = $this->configFactory->get('printable.settings')->get('save_pdf');
+    $pdf_location = $this->configFactory->get('printable.settings')->get('pdf_location');
+    if ($pdf_library == 'wkhtmltopdf') {
+      $this->buildContent();
+      $this->pdfGenerator->setHeader();
+      $this->pdfGenerator->setPageSize($paper_size);
+      $this->pdfGenerator->setPageOrientation($paper_orientation);
+      $this->pdfGenerator->setFooter();
+      if($save_pdf){
+        $filename = $pdf_location;
+        if(empty($filename)){
+          $filename=str_replace("/","_",\Drupal::service('path.current')->getPath());
+          $filename=substr($filename, 1);
+        }
+        $this->pdfGenerator->stream("", $filename.'.pdf');
+      }
+      else
+       $this->pdfGenerator->send();
+    }
+    else if ($pdf_library == 'mPDF'){
+      $this->pdfGenerator->setHeader();
+      $this->pdfGenerator->setPageSize($paper_size);
+      $this->pdfGenerator->setPageOrientation($paper_orientation);
+      $this->pdfGenerator->setFooter();
+      $filename = $pdf_location;
+      $this->mbuildContent($save_pdf,$filename);
+      $this->buildContent();
+    }
+    else {
+      $this->pdfGenerator->setPageOrientation($paper_orientation);
+      $this->buildContent();
+      $this->pdfGenerator->setFooter();
+      if($save_pdf) {
+        $filename = $pdf_location;
+        if(empty($filename)) {
+          $filename=str_replace("/","_",\Drupal::service('path.current')->getPath());
+          $filename=substr($filename, 1);
+        }
+        $this->pdfGenerator->stream("", $filename.'.pdf');
+      }
+      else
+       $this->pdfGenerator->send();
+    }  
+  }
+
+}
```

Comment:
Regarding getResponse()
Place all the library dependent code inside the plugin. Not in this class or method, it should not have knowledge of library specifics on how pdf generator is set up.